c_glib/test/gandiva/test-native-function.rb

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestGandivaNativeFunction < Test::Unit::TestCase
+  def setup
+    omit("Gandiva is required") unless defined?(::Gandiva)
+    @registry = Gandiva::FunctionRegistry.new
+  end
+
+  def test_get_signature
+    assert_kind_of(Gandiva::FunctionSignature,
+                   @registry.native_functions[0].signature)
+  end
+
+  def test_to_string
+    assert_equal(@registry.native_functions[0].signature.to_s,
+                 @registry.native_functions[0].to_s)
+  end
+
+  def test_get_result_nullable_type
+    assert_equal(Gandiva::ResultNullableType::IF_NULL,
+                 @registry.native_functions[0].result_nullable_type)
+
+    isnull_int8 = @registry.native_functions.find do |nf|
+      nf.to_s == "bool isnull(int8)"
+    end
```

Comment:
Umm. We have many `native_functions.find`s.
How about implementing `ggandiva_registry_find_native_function()`?

Reply:
I added `ggandiva_function_registry_lookup_signature` as a wrapper of `gandiva::FunctionRegistry::LookupSignature`.